### PR TITLE
Added more logging for rewrite/redirect scenarios

### PR DESCRIPTION
**Pull request by mikaelm12**
Logging for 
- Redirect to HTTPS
- The current url when continuing to the next url
- The final url when no longer applying rules

@natemcmaster @Tratcher

### CODE REVIEW

File: src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs [hunk 1]
```diff
@@ -75,18 +75,19 @@ public Task Invoke(HttpContext context)
             foreach (var rule in _options.Rules)
             {
                 rule.ApplyRule(rewriteContext);
+                var currentUrl = $"{context.Request.Path + context.Request.QueryString}";
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs]: perf: you only want to do this if logging is enabled

File: src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs [hunk 2]
```diff
@@ -75,18 +75,23 @@ public Task Invoke(HttpContext context)
             foreach (var rule in _options.Rules)
             {
                 rule.ApplyRule(rewriteContext);
+                var currentUrl = string.Empty;
+                if (rewriteContext.Logger.IsEnabled(LogLevel.Debug))
+                {
+                    currentUrl = $"{context.Request.Path + context.Request.QueryString}";
+                }
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs]: nit: newline probable nice here

File: src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs [hunk 3]
```diff
@@ -75,18 +75,23 @@ public Task Invoke(HttpContext context)
             foreach (var rule in _options.Rules)
             {
                 rule.ApplyRule(rewriteContext);
+                var currentUrl = string.Empty;
+                if (rewriteContext.Logger.IsEnabled(LogLevel.Debug))
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs]: Need to check if Logger is null here

File: src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs [hunk 4]
```diff
@@ -75,18 +75,24 @@ public Task Invoke(HttpContext context)
             foreach (var rule in _options.Rules)
             {
                 rule.ApplyRule(rewriteContext);
+                var currentUrl = string.Empty;
+
+                if (rewriteContext.Logger?.IsEnabled(LogLevel.Debug) == true)
+                {
+                    currentUrl = $"{context.Request.Path + context.Request.QueryString}";
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs]: String interpolation is overkill here.

**Comment by natemcmaster** [comment]
As long as you're at it, can we add an information level log message that summarizes the rewrite in one line?

e.g. "Request was rewritten to {newUrl}",

File: src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs [hunk 5]
```diff
@@ -77,10 +77,11 @@ public Task Invoke(HttpContext context)
                 rule.ApplyRule(rewriteContext);
                 var currentUrl = string.Empty;
 
-                if (rewriteContext.Logger?.IsEnabled(LogLevel.Debug) == true)
+                if (rewriteContext.Logger?.IsEnabled(LogLevel.Debug | LogLevel.Information) == true)
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs]: Use `.IsEnabled(LogLevel.Debug) || logger.IsEnabled(LogLevel.Information)` instead. LogLevel isn't designed to be used as bit flags.

File: src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs [hunk 6]
```diff
@@ -97,6 +98,7 @@ public Task Invoke(HttpContext context)
                     default:
                         throw new ArgumentOutOfRangeException($"Invalid rule termination {rewriteContext.Result}");
                 }
+                _logger.RewriteSummary(currentUrl);
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs]: It's possible this logger statement won't be produced if SkipRemainingRules is true. It's also possible that this summary is inaccurate. This should instead go inside the `IRule` implementations that actually rewrite the URL.

**Pull request merged**